core/src/transaction_scheduler.rs

```diff
@@ -0,0 +1,1631 @@
+
+            // already checked we can lock accounts in check_accounts_not_blocked
+            Self::lock_accounts(
+                &mut scheduled_accounts_l,
+                &account_locks.writable,
+                &account_locks.readonly,
+            )
+            .unwrap();
+
+            Ok(sanitized_tx)
+        }
+    }
+
+    fn upsert_higher_fee_account_lock(
+        writable_keys: &[&Pubkey],
+        readonly_keys: &[&Pubkey],
+        highest_wl_blocked_account_fees: &mut HashMap<Pubkey, u64>,
+        highest_rl_blocked_account_fees: &mut HashMap<Pubkey, u64>,
+        priority: u64,
+    ) {
+        for acc in readonly_keys {
+            match highest_rl_blocked_account_fees.entry(**acc) {
+                Entry::Occupied(mut e) => {
+                    if priority > *e.get() {
+                        // NOTE: this should never be the case!
+                        e.insert(priority);
+                    }
+                }
+                Entry::Vacant(e) => {
+                    e.insert(priority);
+                }
+            }
+        }
+
+        for acc in writable_keys {
+            match highest_wl_blocked_account_fees.entry(**acc) {
+                Entry::Occupied(mut e) => {
+                    if priority > *e.get() {
+                        // NOTE: this should never be the case!
+                        e.insert(priority);
+                    }
+                }
+                Entry::Vacant(e) => {
+                    e.insert(priority);
+                }
+            }
+        }
+    }
+
+    fn check_accounts_not_blocked(
+        account_locks: &AccountLocks,
+        highest_wl_blocked_account_fees: &HashMap<Pubkey, u64>,
+        highest_rl_blocked_account_fees: &HashMap<Pubkey, u64>,
+        writable_keys: &[&Pubkey],
+        readonly_keys: &[&Pubkey],
+    ) -> Result<()> {
+        // writes are blocked if there's a blocked read or write account already
+        for acc in writable_keys {
+            if highest_wl_blocked_account_fees.get(*acc).is_some()
+                || highest_rl_blocked_account_fees.get(*acc).is_some()
+            {
+                trace!("write locked account blocked by another tx: {:?}", acc);
+                return Err(SchedulerError::AccountBlocked(**acc));
+            }
+        }
+
+        // reads are blocked if the account exists in write blocked state
+        for acc in readonly_keys {
+            if highest_wl_blocked_account_fees.get(*acc).is_some() {
+                trace!("read locked account blocked by another tx: {:?}", acc);
+                return Err(SchedulerError::AccountBlocked(**acc));
+            }
+        }
+
+        // double check to make sure we can lock this against currently executed transactions and
+        // accounts
+        Self::can_lock_accounts(account_locks, writable_keys, readonly_keys)?;
+
+        Ok(())
+    }
+
+    fn get_scheduled_batch(
```

Comment:
I think this function is worthy of a few comments above declaration. I understand it as:
```
// 1. pop highest priority pkt until requested batch size is reached
// 2. push it onto the sanitized queue if there's not a higher weighted tx scheduled OR currently executing with conflicting state access
// 3. push conflicting lower priority txs onto a reschedule queue
// 4. drop any bad pkts
```